tom_dataproducts/alertstreams/hermes.py

```
@@ -28,43 +29,83 @@ def __init__(self, title='', submitter='', authors='', message='', topic='hermes
         self.extra_info = kwargs
 
 
-def publish_photometry_to_hermes(message_info, datums, **kwargs):
+def publish_to_hermes(message_info, datums, targets=Target.objects.none(), **kwargs):
     """
-    Submits a typical hermes photometry alert using the datums supplied to build a photometry table.
+    Submits a typical hermes alert using the photometry and targets supplied to build a photometry table.
     -- Stores an AlertStreamMessage connected to each datum to show that the datum has previously been shared.
     :param message_info: HERMES Message Object created with BuildHermesMessage
-    :param datums: Queryset of Reduced Datums to be built into table.
+    :param datums: Queryset of Reduced Datums to be built into table. (Will also pull in targets)
+    :param targets: Queryset of Targets to be built into table.
     :return: response
     """
     stream_base_url = settings.DATA_SHARING['hermes']['BASE_URL']
     submit_url = stream_base_url + 'api/v0/' + 'submit_message/'
     # You will need your Hermes API key. This can be found on your Hermes profile page.
     headers = {'Authorization': f"Token {settings.DATA_SHARING['hermes']['HERMES_API_KEY']}"}
 
+    alert = create_hermes_alert(message_info, datums, targets, **kwargs)
+
+    try:
+        response = requests.post(url=submit_url, json=alert, headers=headers)
+        response.raise_for_status()
+        # Only mark the datums as shared if the sharing was successful
+        hermes_alert = AlertStreamMessage(topic=message_info.topic, exchange_status='published')
+        hermes_alert.save()
+        for tomtoolkit_photometry in datums:
+            tomtoolkit_photometry.message.add(hermes_alert)
+    except Exception:
```

Should this be a more specific exception or list of exceptions? 
This seems like it could hide other errors.

---

Hmm I think we probably want to catch any exception so the page doesn't crash. The response is returned so if it's an error response or doesn't contain the message we expect, then we will report the error back to the user.